# Examen 1 (Correciones)

## Teoría

- 1. **¿Qué es un pipeline?** 

Un pipeline es es una serie de pasos que se ejecutan uno tras otro, donde la salida de una fase se convierte en la entrada de la siguiente. Su propósito es mover y transformar datos ordenadamente.

- 2. **¿Cuál es el propósito de realizar regresiones? Explica las ventajas y desventajas de los dos planteamientos vistos en clase.**

La regresión sirve para analizar la relación entre variables y predecir un valor numérico (y) a partir de otro (X).

Ventajas:

-Es fácil interpretar.

-Permite predecir valores futuros.

-Ayuda a identificar qué variables influyen más en el resultado.

Desventajas:

-Es sensible a valores atípicos (outliers).

-Puede haber sobreajuste o problemas de multicolinealidad.


- 3. **¿En qué consiste el proceso de escalamiento de factores?**

Escalar significa poner todas las variables en la misma escala, normalmente entre 0 y 1 o con media 0 y desviación 1. Esto evita que variables con números grandes dominen el modelo.

- 4. **Explica el propósito de penalizar factores en una regresión.**

El propósito de penalizar factores es reducir la importancia de variables poco útiles o irrelevantes. Esto evita que las Betas crezcan demasiado y ayuda a que las predicciones sean más precisas.

- 5. **¿Cuál es la relación entre escalamiento y penalización?**

Para que la penalización funcione bien, las variables deben estar escaladas. Si no, las variables con números más grandes recibirían un castigo más fuerte solo por su tamaño, no por su importancia real.

- 6. **Explica el concepto de una prueba de hipótesis.**

Una prueba de hipótesis es un método estadístico que se usa para decidir, con base en datos, si una afirmación es probablemente cierta o no. Se plantea una hipótesis inicial (H₀) y una alternativa (H₁), y a partir de una muestra se evalúa si hay suficiente evidencia para rechazar H₀.

- 7. **Explica la interpretación de un p-value de una prueba de hipótesis que compara contra una media µ.**

El p-value nos dice la probabilidad de ver los datos que tenemos si la hipótesis que estamos probando es cierta.

Si el p-value es pequeño (ej. < 0.05), los datos no concuerdan con la hipótesis, y podemos rechazarla.
Si el p-value es grande, no hay evidencia suficiente para rechazar la hipótesis.

- 8. **Describe el propósito de realizar cross-validation.**

Cross-validation sirve para probar qué tan bien funciona un modelo en datos nuevos. Se divide el conjunto de datos en partes, se entrena en algunas y se prueba en otras, y así se obtiene una medida más confiable del rendimiento.

- 9. **Describe los pasos que seguirías al hacer un análisis exploratorio de datos. Justifica cada paso.**

-Revisar los datos: entender qué columnas y tipos de datos hay.

-Investigar sobre el tema del dataset para poder interpretar los resultados.

-Limpiar los datos: quitar valores faltantes (Nans) o duplicados.

-Explorar estadísticas básicas: medias, medianas, desviaciones para conocer la distribución de cada variable.

-Visualizar los datos: usar gráficos para ver patrones, relaciones o valores atípicos.


- 10. **¿Qué es el teorema del límite central?**

Es un principio que dice que la media de muchas muestras aleatorias de una población tiende a tener una distribución normal, sin importar la distribución original de los datos. Esto permite usar técnicas estadísticas basadas en la normalidad sin importar la forma original de la población.

## Práctica

In [84]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.metrics import r2_score
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import Ridge, Lasso
from scipy import stats

### 1. Encontrar la parábola que pasa exactamente por los puntos (1, 2), (1,1) y (3,2). Utiliza la librería sklearn.

In [85]:
# Puntos dados
X = np.array([[1],[2],[3]])
y = np.array([2,1,2])

X2 = np.hstack([X, X**2]) 

lr2=LinearRegression() 
lr2.fit(X2,y) 

y_pred2 = lr2.predict(X2) 

print("Intercepto",lr2.intercept_,"Coeficientes",lr2.coef_)

Intercepto 4.9999999999999964 Coeficientes [-4.  1.]


In [86]:
X_ols=sm.add_constant(X2)
ols = sm.OLS(y, X_ols)
results = ols.fit()
results.summary()

C:\Users\Michelle Trigo\anaconda3\Lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 3 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
C:\Users\Michelle Trigo\anaconda3\Lib\site-packages\statsmodels\regression\linear_model.py:1794: RuntimeWarning: divide by zero encountered in divide
  return 1 - (np.divide(self.nobs - self.k_constant, self.df_resid)
C:\Users\Michelle Trigo\anaconda3\Lib\site-packages\statsmodels\regression\linear_model.py:1794: RuntimeWarning: invalid value encountered in scalar multiply
  return 1 - (np.divide(self.nobs - self.k_constant, self.df_resid)
C:\Users\Michelle Trigo\anaconda3\Lib\site-packages\statsmodels\regression\linear_model.py:1716: RuntimeWarning: divide by zero encountered in scalar divide
  return np.dot(wresid, wresid) / self.df_resid


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                      y   R-squared:                       1.000
Model:                            OLS   Adj. R-squared:                    nan
Method:                 Least Squares   F-statistic:                       nan
Date:                Mon, 29 Sep 2025   Prob (F-statistic):                nan
Time:                        21:36:32   Log-Likelihood:                 99.193
No. Observations:                   3   AIC:                            -192.4
Df Residuals:                       0   BIC:                            -195.1
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          5.0000        inf          0        nan         nan         nan
x1            -4.0000        inf         -0        nan         nan         nan
x2             1.0000        inf          0        nan         nan         nan
==============================================================================
Omnibus:                          nan   Durbin-Watson:                   0.059
Prob(Omnibus):                    nan   Jarque-Bera (JB):                0.531
Skew:                          -0.707   Prob(JB):                        0.767
Kurtosis:                       1.500   Cond. No.                         70.9
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

#### Ecuación de la parábola

La ecuación de la parábola es:
$y = x^2 - 4x + 5$.


### 2. Dado el siguiente código, explica paso por paso qué es lo que se está haciendo y el objetivo.

El código carga datos de viviendas en California, separa las características (X) y la variable objetivo (y), normaliza las columnas con StandardScaler y entrena un modelo de regresión lineal dentro de un pipeline. Después aplica un K-Folds Validation, con 10 folds, es decir, divide el dataset en 10 partes iguales, el modelo se entrena 9 veces con 9 partes y se prueba con la parte que quedó fuera. Esto se repite 10 veces, cambiando cada vez la parte que se deja fuera. Al final se tienen 10 resultados de error (MSE), que se promedian. El objetivo es  ver qué tan bien la regresión lineal predice los precios de las casas.




### 3. Entrega un modelo para predecir el "IMDB_Rating" de una película dados los factores que consideres adecuados.

- Utiliza sklearn para el modelo.

- Utiliza escalamiento para los factores numéricos, y transformaciones para los factores cualitativos.

- Calcula los p-values de los factores utilizados. Interpreta tus resultados.

In [87]:
df = pd.read_csv("imdb_top1000_lae.csv")
df.head()

Unnamed: 0              Series_Title Released_Year Certificate  Runtime  \
0           0  The Shawshank Redemption          1994           A  142 min   
1           1             The Godfather          1972           A  175 min   
2           2           The Dark Knight          2008          UA  152 min   
3           3    The Godfather: Part II          1974           A  202 min   
4           4              12 Angry Men          1957           U   96 min   

   IMDB_Rating  Meta_score              Director        Gross  Drama  ...  \
0       9.3000     80.0000        Frank Darabont   28,341,469      1  ...   
1       9.2000    100.0000  Francis Ford Coppola  134,966,411      1  ...   
2       9.0000     84.0000     Christopher Nolan  534,858,444      1  ...   
3       9.0000     90.0000  Francis Ford Coppola   57,300,000      1  ...   
4       9.0000     96.0000          Sidney Lumet    4,360,000      1  ...   

   Fantasy  Family  Thriller  Romance  Sci-Fi  War  Music  Musical  Sport  \
0        0       0         0        0       0    0      0        0      0   
1        0       0         0        0       0    0      0        0      0   
2        0       0         0        0       0    0      0        0      0   
3        0       0         0        0       0    0      0        0      0   
4        0       0         0        0       0    0      0        0      0   

   History  
0        0  
1        0  
2        0  
3        0  
4        0  

[5 rows x 30 columns]

In [88]:
df = df.dropna().drop_duplicates()

In [89]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 714 entries, 0 to 997
Data columns (total 30 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     714 non-null    int64  
 1   Series_Title   714 non-null    object 
 2   Released_Year  714 non-null    object 
 3   Certificate    714 non-null    object 
 4   Runtime        714 non-null    object 
 5   IMDB_Rating    714 non-null    float64
 6   Meta_score     714 non-null    float64
 7   Director       714 non-null    object 
 8   Gross          714 non-null    object 
 9   Drama          714 non-null    int64  
 10  Crime          714 non-null    int64  
 11  Action         714 non-null    int64  
 12  Biography      714 non-null    int64  
 13  Western        714 non-null    int64  
 14  Comedy         714 non-null    int64  
 15  Adventure      714 non-null    int64  
 16  Animation      714 non-null    int64  
 17  Horror         714 non-null    int64  
 18  Mystery   

In [90]:
df.describe()

Unnamed: 0  IMDB_Rating  Meta_score    Drama    Crime   Action  \
count    714.0000     714.0000    714.0000 714.0000 714.0000 714.0000   
mean     518.5728       7.9371     77.1583   0.7003   0.1989   0.1961   
std      295.8481       0.2933     12.4011   0.4585   0.3994   0.3973   
min        0.0000       7.6000     28.0000   0.0000   0.0000   0.0000   
25%      262.2500       7.7000     70.0000   0.0000   0.0000   0.0000   
50%      526.5000       7.9000     78.0000   1.0000   0.0000   0.0000   
75%      777.7500       8.1000     86.0000   1.0000   0.0000   0.0000   
max      997.0000       9.3000    100.0000   1.0000   1.0000   1.0000   

       Biography  Western   Comedy  Adventure  ...  Fantasy   Family  \
count   714.0000 714.0000 714.0000   714.0000  ... 714.0000 714.0000   
mean      0.1232   0.0224   0.2255     0.2283  ...   0.0770   0.0602   
std       0.3290   0.1481   0.4182     0.4200  ...   0.2668   0.2381   
min       0.0000   0.0000   0.0000     0.0000  ...   0.0000   0.0000   
25%       0.0000   0.0000   0.0000     0.0000  ...   0.0000   0.0000   
50%       0.0000   0.0000   0.0000     0.0000  ...   0.0000   0.0000   
75%       0.0000   0.0000   0.0000     0.0000  ...   0.0000   0.0000   
max       1.0000   1.0000   1.0000     1.0000  ...   1.0000   1.0000   

       Thriller  Romance   Sci-Fi      War    Music  Musical    Sport  History  
count  714.0000 714.0000 714.0000 714.0000 714.0000 714.0000 714.0000 714.0000  
mean     0.1387   0.1232   0.0784   0.0406   0.0490   0.0154   0.0238   0.0532  
std      0.3458   0.3290   0.2690   0.1975   0.2161   0.1232   0.1526   0.2246  
min      0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000  
25%      0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000  
50%      0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000  
75%      0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000   0.0000  
max      1.0000   1.0000   1.0000   1.0000   1.0000   1.0000   1.0000   1.0000  

[8 rows x 24 columns]

In [91]:
df['Released_Year'] = pd.to_numeric(df['Released_Year'], errors='coerce')
df = df.dropna(subset=['Released_Year'])
df['Released_Year'] = df['Released_Year'].astype(int)
df['Gross'] = df['Gross'].str.replace(',', '').astype(int)
df['Runtime'] = df['Runtime'].str.replace(' min', '').astype(int)
df['Meta_score'] = df['Meta_score'].astype(int)

**Columnas NO relevantes**
- Series_Title → El nombre de la película no aporta directamente al rating.
- IMDB_Rating → Es la variable objetivo, por lo que no se puede usar de predictor.

In [92]:
df = df.drop(columns=["Series_Title","Director"])
X = df.drop(columns=["IMDB_Rating"])
y = df["IMDB_Rating"]

In [93]:
X=pd.get_dummies(X, columns=["Certificate"])

In [94]:
Xtrain, Xtest, ytrain, ytest=train_test_split(X,y,train_size=.4,random_state=137)

In [95]:
scaler = StandardScaler().fit(Xtrain)
Xtrain_scaled = scaler.transform(Xtrain)
Xtest_scaled  = scaler.transform(Xtest)

In [96]:
X_ols=sm.add_constant(Xtrain_scaled)
ols = sm.OLS(ytrain, X_ols)
results = ols.fit()
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:            IMDB_Rating   R-squared:                       0.886
Model:                            OLS   Adj. R-squared:                  0.870
Method:                 Least Squares   F-statistic:                     53.78
Date:                Mon, 29 Sep 2025   Prob (F-statistic):           3.24e-97
Time:                        21:36:35   Log-Likelihood:                 255.09
No. Observations:                 285   AIC:                            -436.2
Df Residuals:                     248   BIC:                            -301.0
Df Model:                          36                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          7.9312      0.006   1263.355      0.000       7.919       7.944
x1            -0.2658      0.007    -36.635      0.000      -0.280      -0.251
x2            -0.0380      0.009     -4.187      0.000      -0.056      -0.020
x3             0.0191      0.008      2.284      0.023       0.003       0.035
x4             0.0043      0.007      0.592      0.554      -0.010       0.019
x5             0.0107      0.008      1.285      0.200      -0.006       0.027
x6             0.0088      0.011      0.837      0.403      -0.012       0.030
x7            -0.0001      0.008     -0.017      0.987      -0.016       0.016
x8             0.0002      0.009      0.022      0.982      -0.017       0.017
x9            -0.0112      0.007     -1.500      0.135      -0.026       0.004
x10           -0.0073      0.007     -1.051      0.294      -0.021       0.006
x11           -0.0065      0.008     -0.787      0.432      -0.023       0.010
x12           -0.0156      0.010     -1.542      0.124      -0.035       0.004
x13            0.0072      0.009      0.812      0.417      -0.010       0.025
x14            0.0059      0.008      0.745      0.457      -0.010       0.022
x15           -0.0071      0.007     -0.950      0.343      -0.022       0.008
x16           -0.0075      0.008     -0.928      0.354      -0.023       0.008
x17           -0.0017      0.007     -0.229      0.819      -0.016       0.013
x18            0.0068      0.007      0.914      0.362      -0.008       0.022
x19           -0.0094      0.008     -1.201      0.231      -0.025       0.006
x20           -0.0148      0.008     -1.971      0.050      -0.030   -8.92e-06
x21           -0.0006      0.008     -0.074      0.941      -0.016       0.015
x22           -0.0063      0.007     -0.887      0.376      -0.020       0.008
x23           -0.0045      0.008     -0.559      0.577      -0.020       0.011
x24           -0.0055      0.009     -0.615      0.539      -0.023       0.012
x25           -0.0099      0.007     -1.394      0.165      -0.024       0.004
x26           -0.0050      0.007     -0.732      0.465      -0.019       0.008
x27            0.0150      0.005      2.796      0.006       0.004       0.026
x28           -0.0068      0.008     -0.849      0.397      -0.023       0.009
x29           -0.0046      0.008     -0.562      0.575      -0.021       0.011
x30            0.0067      0.007      0.960      0.338      -0.007       0.020
x31           -0.0028      0.007     -0.395      0.693      -0.017       0.011
x32            0.0003      0.006      0.051      0.960      -0.012       0.013
x33           -0.0125      0.008     -1.619      0.107      -0.028       0.003
x34           -0.0046      0.006     -0.784      0.434      -0.016       0.007
x35           -0.0040      0.006     -0.626      0.532      -0.017       0.009
x3

In [97]:
Xtrain_scaled_const = sm.add_constant(Xtrain_scaled, has_constant="add")
Xtest_scaled_const  = sm.add_constant(Xtest_scaled, has_constant="add")

y_pred_train = results.predict(Xtrain_scaled_const)
y_pred_test  = results.predict(Xtest_scaled_const)

print("R2 train:", r2_score(ytrain, y_pred_train))
print("R2 test:",  r2_score(ytest, y_pred_test))


R2 train: 0.8864486792366304
R2 test: 0.8967342420165834


1. Ajuste general del modelo

El modelo tiene un R² = 0.886 y un Adj. R² = 0.870, lo que indica que explica una gran parte de la variabilidad de las calificaciones de IMDb. El desempeño en entrenamiento (R² = 0.886) y en prueba (R² = 0.897) es bastante consistente, lo que sugiere que el modelo generaliza bien y no hay un sobreajuste marcado.

2. Variables con impacto significativo

Las variables más relevantes (p < 0.05) son:

Constante: 11.40, p < 0.001

Años de lanzamiento (Released_Year): coef -0.0021, p < 0.001 → películas más recientes ligeramente peor calificadas.

Runtime: coef 0.0007, p = 0.023 → películas más largas tienden a tener un puntaje ligeramente mayor.

Certificados (ratings): Certificate_A, Approved, G, GP, PG, PG-13, Passed, R, TV-PG, U, UA → todos significativos, coeficientes positivos ~0.93 a 1.17, lo que indica que ciertos ratings se asocian con calificaciones más altas.

3. Variables poco o nada significativas

La mayoría de los géneros (Drama, Crime, Action, Adventure, Horror, Sci-Fi, etc.) no son significativos, lo que indica que el género de la película no tiene un efecto fuerte en las calificaciones.

Gross y Meta_score tampoco son significativos, por lo que la recaudación o la puntuación crítica no influyen notablemente en este modelo.

4. Posibles problemas

Muchísimas variables dummies de directores y géneros.


## P-values

In [98]:
lr=LinearRegression()
lr.fit(Xtrain_scaled,ytrain)

y_pred = lr.predict(Xtrain_scaled)

In [99]:
import numpy as np
import pandas as pd
from scipy import stats
from sklearn.linear_model import LinearRegression

# Ajuste del modelo
lr = LinearRegression()
lr.fit(Xtrain_scaled, ytrain)

# Predicciones en train
y_hat = lr.predict(Xtrain_scaled)

# Diseño con constante
X_design = np.column_stack([np.ones(Xtrain_scaled.shape[0]), Xtrain_scaled])

# Residual sum of squares y residual standard error
rss = np.sum((ytrain - y_hat) ** 2)
n, k = Xtrain_scaled.shape
rse = np.sqrt(rss / (n - k - 1))

# Varianzas usando pseudo-inversa en lugar de inversa
var_beta = np.linalg.pinv(X_design.T @ X_design) * rse**2
std_beta = np.sqrt(np.diag(var_beta))

# Coeficientes (intercepto + betas)
coefs = np.insert(lr.coef_, 0, lr.intercept_)

# Estadísticos t y p-values
t_stats = coefs / std_beta
p_values = [2 * (1 - stats.t.cdf(np.abs(t), df=n - k - 1)) for t in t_stats]

# Resultados en un DataFrame
results = pd.DataFrame({
    "Variable": ["Intercepto"] + list(X.columns),
    "Coef": coefs,
    "Std_Error": std_beta,
    "t_value": t_stats,
    "p_value": p_values
})

pd.set_option("display.float_format", lambda x: f"{x:.4f}")
print("\nResultados OLS (todas las variables):")
print(results)



Resultados OLS (todas las variables):
                Variable             Coef  Std_Error            t_value  \
0             Intercepto           7.9312     0.0063          1258.2547   
1             Unnamed: 0          -0.2658     0.0073           -36.4873   
2          Released_Year          -0.0380     0.0091            -4.1698   
3                Runtime           0.0191     0.0084             2.2761   
4             Meta_score           0.0043     0.0073             0.5906   
5                  Gross           0.0107     0.0084             1.2802   
6                  Drama           0.0088     0.0106             0.8326   
7                  Crime          -0.0001     0.0080            -0.0165   
8                 Action           0.0002     0.0086             0.0223   
9              Biography          -0.0112     0.0075            -1.4937   
10               Western          -0.0073     0.0070            -1.0467   
11                Comedy          -0.0065     0.0083         

C:\Users\Michelle Trigo\AppData\Local\Temp\ipykernel_6136\1416078757.py:29: RuntimeWarning: invalid value encountered in divide
  t_stats = coefs / std_beta


1. 
**¿Cuál fue el error?**
Eliminé la columna de Unnamed 0
**¿Cuál es la corrección?**
La incluí en la regresión
**¿Por qué se cometió el error?**
Pensé que no tenía impacto.
**¿Cómo se puede evitar este error en el futuro? (Métodos prácticos aplicables. Nada de “estudiar más”)**
Hacer pruebas y analizar el P-value de todas las variables antes de eliminar alguna.
 
2.  
**¿Cuál fue el error?**
Eliminé la columna de Gross
**¿Cuál es la corrección?**
Modifiqué el df para eliminar las comas de los valores y los pudiera convertir a int.
**¿Por qué se cometió el error?**
No recordé como hacerlo en el momento.
**¿Cómo se puede evitar este error en el futuro? (Métodos prácticos aplicables. Nada de “estudiar más”)**
Recordar técnicas de manipulación de datos básicos, llendo a asesorías o haciendo las prácticas de las asesorías enfocadas en eso.

3.
**¿Cuál fue el error?**
Eliminé la columna de Runtime.
**¿Cuál es la corrección?**
Modifiqué el df para quitarle el "min" de los valores para poder convertirlos a int.
**¿Por qué se cometió el error?**
No recordé como hacerlo en el momento.
**¿Cómo se puede evitar este error en el futuro? (Métodos prácticos aplicables. Nada de “estudiar más”)**
Recordar técnicas de manipulación de datos básicos, llendo a asesorías o haciendo las prácticas de las asesorías enfocadas en eso.

4.
**¿Cuál fue el error?**
Eliminé la columna de Runtime.
**¿Cuál es la corrección?**
Modifiqué el df para quitarle el "min" de los valores para poder convertirlos a int.
**¿Por qué se cometió el error?**
No recordé como hacerlo en el momento.
**¿Cómo se puede evitar este error en el futuro? (Métodos prácticos aplicables. Nada de “estudiar más”)**
Recordar técnicas de manipulación de datos básicos, llendo a asesorías o haciendo las prácticas de las asesorías enfocadas en eso.

5. 
**¿Cuál fue el error?**
No haber borrado la columna de "Director"
**¿Cuál es la corrección?**
Eliminarla antes de hacer la regresión
**¿Por qué se cometió el error?**
No me di cuenta que la mayoría de los directores solo hicieron una película, lo que hace que esta información sea muy específica de cada caso y no generalice bien. Mantenerla causaba que el modelo se sobreajustara a los datos de entrenamiento y fallara al predecir nuevas películas.
**¿Cómo se puede evitar este error en el futuro?** 
En este caso, me hizo falta más tiempo para hacer más pruebas y notar ese punto antes.

6.
**¿Cuál fue el error?**
No haber calculado los p-values a mano.
**¿Cuál es la corrección?**
Calcularlos.
**¿Por qué se cometió el error?**
No me dio suficiente tiempo para hacerlo, aparte no tenía preparado el código para hacerlo.
**¿Cómo se puede evitar este error en el futuro?** 
En este caso, me hizo falta más tiempo, pero hubiera servido tener los códigos preparados desde antes.

7.
**¿Cuál fue el error?**
Interpretación de resultados
**¿Cuál es la corrección?**
Nueva interpretación con los nuevos resultados.
**¿Por qué se cometió el error?**
Interpreté los resultados que tenía, que eran erróneos.
**¿Cómo se puede evitar este error en el futuro?** 
No equivocarme con los resultados para interpretar los correctos.